src/AcmTable/AcmTable.tsx

```diff
@@ -712,6 +770,136 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
 
     const renderCustomTableActionResults = customTableAction && renderCustomTableAction(customTableAction)
 
+    const onFilterSelect = (selection: string | SelectOptionObject) => {
```

Comment:
onFilterSelect should not be going though all filters... it should be called in context of a single filter
so instead of `filterItems.forEach` it should take in a  (selection: string | SelectOptionObject, filter: ITableFIlter)